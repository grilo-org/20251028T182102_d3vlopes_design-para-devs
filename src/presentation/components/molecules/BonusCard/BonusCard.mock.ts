import { generateID } from '@/utils/generateID';

import { BonusModel } from '@/domain/models/Bonus';

export const bonusCardMock: BonusModel[] = [
	{
		id: generateID(),
		icon: {
			src: 'https://media.graphassets.com/bfjH5gwbSZ24nF8iEe21',
			alt: 'Ícone do Discord',
		},
		title: 'Comunidade exclusiva',
		description:
			'Uma comunidade no Discord para conhecer outros devs, trocar conhecimentos e tirar dúvidas.',
		benefitTitle: 'Nesta comunidade, você pode',
		benefits: [
			{
				id: generateID(),
				description: 'Fazer networking',
			},
			{
				id: generateID(),
				description: 'Divulgar seus projetos',
			},
			{
				id: generateID(),
				description: 'Obter análise do seu código por mim',
			},
			{
				id: generateID(),
				description: 'Tirar dúvidas',
			},
		],
	},
	{
		id: generateID(),
		icon: {
			src: 'https://media.graphassets.com/o3ZF4RCMS16tFRMXck2l',
			alt: 'Ícone do Notion',
		},
		title: 'Lista de Recursos',
		description:
			'Acesso a uma lista crescente de recursos no Notion para você aprender ainda mais sobre design.',
		benefitTitle: 'Essa lista contém',
		benefits: [
			{
				id: generateID(),
				description: 'Design Systems',
			},
			{
				id: generateID(),
				description: 'Artigos',
			},
			{
				id: generateID(),
				description: 'Ferramentas',
			},
			{
				id: generateID(),
				description: 'Case Study',
			},
		],
	},
	{
		id: generateID(),
		icon: {
			src: 'https://media.graphassets.com/oBpHb7PQduZBg7Dv7K7A',
			alt: 'Ícone do Figma junto com o ícone do Storybook',
		},
		title: 'Conteúdos complementares',
		description:
			'Tenha acesso a conteúdos complementares relacionados ao assunto principal do eBook para elevar ainda mais suas habilidades.',
		benefitTitle: 'Você vai aprender',
		benefits: [
			{
				id: generateID(),
				description: 'Automação com plop.js',
			},
			{
				id: generateID(),
				description: 'Figma para Devs',
			},
			{
				id: generateID(),
				description: 'Integrando Figma ao Storybook',
			},
		],
	},
];
