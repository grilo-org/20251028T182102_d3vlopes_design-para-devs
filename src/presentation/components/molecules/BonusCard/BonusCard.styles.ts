import { tv } from 'tailwind-variants';

const styles = tv({
	slots: {
		wrapperStyles:
			'flex flex-col py-8 rounded-[4px] border border-solid border-background-200 bg-background-400 h-[42.25rem] md:py-14 md:h-[43.125rem]',
		headerStyles: 'flex flex-col items-center',
		iconStyles:
			'w-max-[11.4375rem] min-w-[5.9375rem] h-[5.9375rem] md:w-max-[17.6875rem] md:w-min-[9.0625rem] md:h-[9.0625rem]',
		titleStyles:
			'my-6 text-center font-highlight font-extrabold text-3xl text-complementary-white max-w-[14.0625rem] md:text-6xl md:max-w-[31.6875rem]',
		dividerStyles: 'border border-solid border-background-200',
		contentWrapperStyles: 'px-3 md:pl-6 md:pr-0',
		benefitDescriptionStyles:
			'font-body text-base text-neutral-50 mt-8 mb-6 max-w-max md:max-w-[35.1875rem]',
		benefitTitleStyles:
			'font-body text-base text-neutral-50 font-semibold mb-4 md:text-lg',
		benefitsWrapper: 'flex items-center gap-2',
		benefitsDescription:
			'font-body text-base leading-[1.3619rem] font-bold text-neutral-50 max-w-[12.4375rem] md:max-w-max',
	},
});

const makeStyles = styles();

export const classNames = {
	wrapperStyles: makeStyles.wrapperStyles(),
	headerStyles: makeStyles.headerStyles(),
	iconStyles: makeStyles.iconStyles(),
	titleStyles: makeStyles.titleStyles(),
	dividerStyles: makeStyles.dividerStyles(),
	contentWrapperStyles: makeStyles.contentWrapperStyles(),
	benefitDescriptionStyles: makeStyles.benefitDescriptionStyles(),
	benefitTitleStyles: makeStyles.benefitTitleStyles(),
	benefitsWrapper: makeStyles.benefitsWrapper(),
	benefitsDescription: makeStyles.benefitsDescription(),
};
