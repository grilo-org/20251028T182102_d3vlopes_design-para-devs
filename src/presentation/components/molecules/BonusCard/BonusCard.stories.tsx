import type { Meta, StoryObj } from '@storybook/react';

import { bonusCardMock } from './BonusCard.mock';

import { BonusCard } from '.';

const meta: Meta<typeof BonusCard> = {
	title: 'Molecules/BonusCard',
	component: BonusCard,
	args: bonusCardMock[0],
	parameters: {
		layout: 'fullscreen',
		design: {
			type: 'figma',
			url: 'https://www.figma.com/file/p1r0o56xHSpG3YHnrNlO3P/LP-Design-para-Devs?type=design&node-id=325-400&mode=design&t=Z5d8zymNiqpqaMmc-0',
		},
	},
};

export default meta;
type Story = StoryObj<typeof BonusCard>;

const template: Story = {
	render: (args) => (
		<div className="max-w-[612px] m-auto p-8">
			<BonusCard {...args} />
		</div>
	),
};

export const Default: Story = {
	...template,
};
