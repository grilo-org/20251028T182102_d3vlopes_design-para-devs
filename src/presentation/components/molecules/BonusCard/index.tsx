import { BonusModel } from '@/domain/models/Bonus';

import { Heading } from '@/presentation/components/atoms/Heading';

import { BenefitPartial } from './partials/BenefitPartial';

import { classNames } from './BonusCard.styles';

export type BonusCardProps = BonusModel;

export const BonusCard = ({
	icon,
	title,
	description,
	benefitTitle,
	benefits,
}: BonusCardProps) => {
	return (
		<div className={classNames.wrapperStyles}>
			<header className={classNames.headerStyles}>
				<img src={icon.src} alt={icon.alt} className={classNames.iconStyles} />

				<Heading>
					<h2 className={classNames.titleStyles}>{title}</h2>
				</Heading>
			</header>

			<hr className={classNames.dividerStyles} />

			<div className={classNames.contentWrapperStyles}>
				<p className={classNames.benefitDescriptionStyles}>{description}</p>

				<Heading>
					<h3 className={classNames.benefitTitleStyles}>{benefitTitle}:</h3>
				</Heading>

				{benefits.map(BenefitPartial)}
			</div>
		</div>
	);
};
