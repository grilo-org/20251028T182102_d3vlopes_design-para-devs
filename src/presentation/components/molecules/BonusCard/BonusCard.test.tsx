import { render, screen } from '@testing-library/react';

import { bonusCardMock } from './BonusCard.mock';

import { BonusCard } from '.';

const { getByRole, getByText, getAllByRole } = screen;

const mock = bonusCardMock[0];

describe('<BonusCard />', () => {
	beforeEach(() => render(<BonusCard {...mock} />));

	it('should render icon', () => {
		const icon = getByRole('img', { name: mock.icon.alt });

		expect(icon).toHaveAttribute('src', mock.icon.src);
	});

	it('should render title', () => {
		const title = getByRole('heading', { name: mock.title });

		expect(title).toBeInTheDocument();
	});

	it('should render description', () => {
		const description = getByText(mock.description);

		expect(description);
	});

	it('should render benefit title', () => {
		const benefitTitle = getByRole('heading', {
			name: `${mock.benefitTitle}:`,
		});

		expect(benefitTitle).toBeInTheDocument();
	});

	it('should render correct benefits icon length', () => {
		const iconsChecked = getAllByRole('img', { name: 'Ícone de positivo' });

		expect(iconsChecked).toHaveLength(mock.benefits.length);
	});

	it('should render benefits description', () => {
		for (const benefit of mock.benefits) {
			const benefitDescription = getByText(benefit.description);

			expect(benefitDescription).toBeInTheDocument();
		}
	});

	it('should match snapshot', () => {
		const { container } = render(<BonusCard {...mock} />);

		expect(container).toMatchSnapshot();
	});
});
