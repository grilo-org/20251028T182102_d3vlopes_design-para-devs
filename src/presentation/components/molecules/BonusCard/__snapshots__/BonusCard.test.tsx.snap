// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BonusCard /> should match snapshot 1`] = `
<div>
  <div
    class="flex flex-col py-8 rounded-[4px] border border-solid border-background-200 bg-background-400 h-[42.25rem] md:py-14 md:h-[43.125rem]"
  >
    <header
      class="flex flex-col items-center"
    >
      <img
        alt="Ícone do Discord"
        class="w-max-[11.4375rem] min-w-[5.9375rem] h-[5.9375rem] md:w-max-[17.6875rem] md:w-min-[9.0625rem] md:h-[9.0625rem]"
        src="https://media.graphassets.com/bfjH5gwbSZ24nF8iEe21"
      />
      <h2
        class="my-6 text-center font-highlight font-extrabold text-3xl text-complementary-white max-w-[14.0625rem] md:text-6xl md:max-w-[31.6875rem]"
      >
        Comunidade exclusiva
      </h2>
    </header>
    <hr
      class="border border-solid border-background-200"
    />
    <div
      class="px-3 md:pl-6 md:pr-0"
    >
      <p
        class="font-body text-base text-neutral-50 mt-8 mb-6 max-w-max md:max-w-[35.1875rem]"
      >
        Uma comunidade no Discord para conhecer outros devs, trocar conhecimentos e tirar dúvidas.
      </p>
      <h3
        class="font-body text-base text-neutral-50 font-semibold mb-4 md:text-lg"
      >
        Nesta comunidade, você pode
        :
      </h3>
      <div
        class="flex items-center gap-2"
      >
        <img
          alt="Ícone de positivo"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2Fd65K9uHRScCtAVILVcY8&w=96&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2Fd65K9uHRScCtAVILVcY8&w=48&q=75 1x, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2Fd65K9uHRScCtAVILVcY8&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
        <span
          class="font-body text-base leading-[1.3619rem] font-bold text-neutral-50 max-w-[12.4375rem] md:max-w-max"
        >
          Fazer networking
        </span>
      </div>
      <div
        class="flex items-center gap-2"
      >
        <img
          alt="Ícone de positivo"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2Fd65K9uHRScCtAVILVcY8&w=96&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2Fd65K9uHRScCtAVILVcY8&w=48&q=75 1x, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2Fd65K9uHRScCtAVILVcY8&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
        <span
          class="font-body text-base leading-[1.3619rem] font-bold text-neutral-50 max-w-[12.4375rem] md:max-w-max"
        >
          Divulgar seus projetos
        </span>
      </div>
      <div
        class="flex items-center gap-2"
      >
        <img
          alt="Ícone de positivo"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2Fd65K9uHRScCtAVILVcY8&w=96&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2Fd65K9uHRScCtAVILVcY8&w=48&q=75 1x, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2Fd65K9uHRScCtAVILVcY8&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
        <span
          class="font-body text-base leading-[1.3619rem] font-bold text-neutral-50 max-w-[12.4375rem] md:max-w-max"
        >
          Obter análise do seu código por mim
        </span>
      </div>
      <div
        class="flex items-center gap-2"
      >
        <img
          alt="Ícone de positivo"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2Fd65K9uHRScCtAVILVcY8&w=96&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2Fd65K9uHRScCtAVILVcY8&w=48&q=75 1x, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2Fd65K9uHRScCtAVILVcY8&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
        <span
          class="font-body text-base leading-[1.3619rem] font-bold text-neutral-50 max-w-[12.4375rem] md:max-w-max"
        >
          Tirar dúvidas
        </span>
      </div>
    </div>
  </div>
</div>
`;
